src/polyswarmd/event_message.py

```diff
@@ -87,56 +84,16 @@ def channel_events(ws, guid):
         msig_address = offer_channel['msig_address']
         offer_msig = g.chain.offer_multisig.bind(msig_address)
+        fmanager = FilterManager()
+        for evt in [ClosedAgreement, StartedSettle, SettleStateChallenged]:
+            fmanager.register(offer_msig.eventFilter(evt.filter_event), evt)
 
-        filters_initialized = False
-        while not ws.closed:
-            try:
-                if not filters_initialized:
-                    closed_agreement_filter = offer_msig.eventFilter('ClosedAgreement')
-                    settle_started_filter = offer_msig.eventFilter('StartedSettle')
-                    settle_challenged_filter = offer_msig.eventFilter('SettleStateChallenged')
-
-                    filters_initialized = True
-
-                for event in closed_agreement_filter.get_new_entries():
-                    ws.send(
-                        json.dumps({
-                            'event': 'closed_agreement',
-                            'data': new_cancel_agreement_event_to_dict(event.args),
-                            'block_number': event.blockNumber,
-                            'txhash': event.transactionHash.hex(),
-                        }))
-
-                for event in settle_started_filter.get_new_entries():
-                    ws.send(
-                        json.dumps({
-                            'event': 'settle_started',
-                            'data': new_settle_started_event(event.args),
-                            'block_number': event.blockNumber,
-                            'txhash': event.transactionHash.hex(),
-                        }))
-
-                for event in settle_challenged_filter.get_new_entries():
-                    ws.send(
-                        json.dumps({
-                            'event': 'settle_challenged',
-                            'data': new_settle_challenged_event(event.args),
-                            'block_number': event.blockNumber,
-                            'txhash': event.transactionHash.hex(),
-                        }))
+        def send(msg):
+            if ws.closed:
+                raise RuntimeError("WebSocket is closed")
+            return ws.send(msg)
 
-                gevent.sleep(1)
-            except WebSocketError:
-                logger.info('Websocket connection closed, exiting loop')
-                break
-            except ConnectionError:
-                logger.exception('ConnectionError in offer /events (is geth down?)')
-                filters_initialized = False
-                continue
-            except Exception:
-                logger.exception('Exception in /events, resetting filters')
-                filters_initialized = False
-                continue
+        fmanager.event_pool(ws.send).join()
 
     # for receiving messages about offers that might need to be signed
     @sockets.route('/messages/<uuid:guid>')
```

Comment:
Should we just purge this @Rizato ? I don't like parts of this code as it is, anyway.

Reply:
Yea, but not this PR.